### ISSUE TITLE
LR Tuner suggest `min/max` learning rate for cyclic schedulers

### ISSUE THREAD

**Issue by RuRo** [PyTorchLightning/pytorch-lightning#10089]
## 🚀 Feature

Better support for using Cyclic learning rate schedulers (`OneCycleLR`, `CyclicLR`, `CosineAnnealing*LR`) with the LR Tuner.

1) Add an optional `cyclic=False` argument to [`_LRFinder.suggestion` and `_LRFinder.plot`](https://github.com/PyTorchLightning/pytorch-lightning/blob/495aa44fb7e081a50c06a9be0403adfbaeb4c136/pytorch_lightning/tuner/lr_finder.py#L173). When it is set to `True`, `suggestion` will return a tuple `(base_lr, max_lr)` - the recommended learning rate bounds for cyclic learning rates.

2) Change [`lr_find`](https://github.com/PyTorchLightning/pytorch-lightning/blob/495aa44fb7e081a50c06a9be0403adfbaeb4c136/pytorch_lightning/tuner/lr_finder.py#L195), `Trainer` and `Tuner` to use the above optional argument, if `auto_lr_find` is a tuple of strings (e.g. `("base_lr", "max_lr")`). This will set these 2 keys in `hparams` after tuning.

This will allow one to write something like

```python
def configure_optimizers(self):
    ...
    optimizer = SomeOptimizer(..., lr=self.hparams.base_lr)
    scheduler = CyclicLR(optimizer, self.hparams.base_lr, self.hparams.max_lr)
    ... 

trainer = Trainer(..., auto_lr_find=("base_lr", "max_lr"))
```

### Motivation

According to the documentation, the LR Tuner is based on [this paper](https://arxiv.org/abs/1506.01186), however, the method in the paper is actually a method for tuning the **learning rate bounds of cyclic learning rate schedulers** (See section 3.3 "How can one estimate reasonable minimum and maximum boundary values?"). And yet, this use case is not currently supported out of the box.

Currently, the implementation of `suggestion` just returns the lower bound (`base_lr`), which might work for non-cyclic schedulers, but isn't sufficient for cyclic schedulers. Even worse, with the way the arguments of `OneCycleLR` and `CosineAnnealing*LR` are structured, the most intuitive way to use them (aka just plug in all the required arguments with the closest matches) actually gives **wrong** results when used with the LR Tuner:

```python
def configure_optimizers(self):
    ...
    optimizer = SomeOptimizer(..., lr=self.hparams.lr)
    scheduler = OneCycleLR(optimizer, self.hparams.lr)
    # or
    scheduler = CosineAnnealingLR(optimizer, self.trainer.max_epochs)
    ... 

trainer = Trainer(..., auto_lr_find=True)
```

This is wrong, since the `lr` that is set by the tuner is intended to be used as the `base_lr` value, but in the above example, it ends up being used as the `max_lr` value, while the `base_lr` is inferred from it or set to 0.

cc @borda

**Comment by RuRo** [comment]
I would like to keep this issue alive until someone from the pytorch-lightning team takes a look at it. 😄 

Also, I should mention, that I am willing to spend some time making a PR for this feature myself (if/after somebody from the maintainer team OKs my proposal).

**Comment by RuRo** [comment]
bump

**Comment by grudloff** [comment]
This is an interesting proposal!
